src/main/java/com/aws/greengrass/logmanager/model/LogFileGroup.java

```diff
@@ -42,11 +56,13 @@ public static LogFileGroup create(Pattern filePattern, URI path, Instant lastUpd
         List<LogFile> allFiles = new ArrayList<>();
         if (files.length != 0) {
             for (LogFile file: files) {
+                String fileHash = file.hashString();
                 if (file.isFile()
                         && lastUpdated.isBefore(Instant.ofEpochMilli(file.lastModified()))
                         && filePattern.matcher(file.getName()).find()
-                        && file.length() > 0) {
+                        && !file.isEmpty(fileHash)) {
                     allFiles.add(file);
+                    fileHashToFileMap.put(fileHash, file);
```

Comment:
Why have 2 data structures holding the same stuff?. You could do away with just the map and keep track of the active file on a separate variable basically while we loop through the files get check the lastModified and store the one with the greatest value.

Reply:
This map is not for active file. It is for getting the file to be deleted.

Reply:
Yeah, I know I am saying we don't need 2 data structures we could do away with that map and a variable that stores the active file

Reply:
Actually, besides the reason of getting the active file from sorted files, in case of some edge cases, the LM wants to process the oldest file first. E.g. when processing new file, the older files get deleted due to the disk space limit, so LM need to process older files first to avoid data loss.